src/app/models/__tests__/form.server.model.spec.ts

```diff
@@ -1202,6 +1203,131 @@ describe('Form Model', () => {
         })
       })
     })
+
+    describe('updateFormLogic', () => {
```

Comment:
should we have a test for when the logic ID doesn't exist? just to be defensive, I know that it's already validated in the service, but there may be very rare race conditions where the logic gets deleted after the service check but before the db update

Reply:
thanks for pointing out - in this case, the DB would return an unmodified form, and at the service level, we check if the logic exists in the returned form and if not, then return logicNotFoundError i.e. the logic was deleted in between the service check and db update